lib/srv/desktop/tdp/conn.go

```diff
@@ -164,3 +167,117 @@ func IsFatalErr(err error) bool {
 
 	return !IsNonFatalErr(err)
 }
+
+// ProxyConn does a bidirectional copy between the connection to the browser/Connect
+// (client) and the mTLS connection to Windows Desktop Service (server).
+func ProxyConn(ctx context.Context, client, server io.ReadWriteCloser) error {
```

Comment:
I noticed that Przemko merged a latency detector for desktop access, so it'll probably be a bit of a pain to get rebase/merge master to this branch. #52827

I'd say it'd be probably best if you asked him to look through the extraction of `ProxyConn` after you're done.

Reply:
@probakowski @ravicious could I ask you for a final look at the commit https://github.com/gravitational/teleport/pull/54373/commits/cc5f4fee7bdb7e43d488dca354d49a3fb990a044?
Merging master to this branch was indeed a pain 🙈 
I adapted the code in the `tdp/conn.go` to the changes from master (err group), but I left the latency monitor code in `web/desktop.go` (it can't work in Connect anyways). I also made some improvements, I believe we didn't properly handle context cancellation before.

To support the latency detector, I allowed passing an interceptor to the proxy code, so it's possible to inspect messages coming from the server (it will be useful for directory sharing in Connect too).